File: src/com/laamella/intellij6502/asm/AssemblyCommenter.java
package com.laamella.intellij6502.asm;

import com.intellij.lang.Commenter;
import org.jetbrains.annotations.Nullable;

public class AssemblyCommenter implements Commenter {
    @Nullable
    @Override
    public String getLineCommentPrefix() {
        return ";";
    }

    @Nullable
    @Override
    public String getBlockCommentPrefix() {
        return ".comment";
    }

    @Nullable
    @Override
    public String getBlockCommentSuffix() {
        return ".endc";
    }

    @Nullable
    @Override
    public String getCommentedBlockCommentPrefix() {
        return null;
    }

    @Nullable
    @Override
    public String getCommentedBlockCommentSuffix() {
        return null;
    }
}

File: src/main/kotlin/com/laamella/intellij6502/asm/AssemblyCommenter.kt
package com.laamella.intellij6502.asm

import com.intellij.lang.Commenter

class AssemblyCommenter : Commenter {
    override fun getLineCommentPrefix(): String? {
        return ";"
    }

    override fun getBlockCommentPrefix(): String? {
        return ".comment"
    }

    override fun getBlockCommentSuffix(): String? {
        return ".endc"
    }

    override fun getCommentedBlockCommentPrefix(): String? {
        return null
    }

    override fun getCommentedBlockCommentSuffix(): String? {
        return null
    }
}